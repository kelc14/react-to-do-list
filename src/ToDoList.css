@import url("https://fonts.googleapis.com/css2?family=Roboto+Slab:wght@300;500&display=swap");

.ToDoList {
  margin: 0 auto;
  width: 70vw;
  min-height: 40vh;
  text-align: center;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
}
.ToDoList-header {
  padding-top: 25px;
  font-family: "Roboto Slab", serif;
  font-weight: 800;
}
.ToDoList-form {
  font-family: "Roboto Slab", serif;
  font-weight: 300;
  margin-bottom: 15px;
}
.ToDoList-list-container {
  margin-top: 30px;
  width: 62vw;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
}
