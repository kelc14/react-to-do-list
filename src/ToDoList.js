import React, { useState } from "react";
import "./ToDoList.css";
import ToDo from "./ToDo";
import ToDoForm from "./ToDoForm";
import { v4 as uuid } from "uuid";

const ToDoList = () => {
  const [toDos, setToDos] = useState([]);

  const addItem = (toDo) => {
    let newToDo = { ...toDo, id: uuid() };
    setToDos((toDos) => [...toDos, newToDo]);
  };

  /** Remove the selected item from state. */
  const removeItem = (itemId) => {
    let noItem = toDos.filter((el) => el.id !== itemId);
    setToDos(() => [...noItem]);
  };

  return (
    <div className="ToDoList">
      <h2 className="ToDoList-header">My To Do List</h2>
      <div className="ToDoList-form">
        <ToDoForm addItem={addItem} />
      </div>
      <div className="ToDoList-list-container">
        {toDos.map((n) => (
          <ToDo name={n.name} key={n.id} id={n.id} removeItem={removeItem} />
        ))}
      </div>
    </div>
  );
};

export default ToDoList;
