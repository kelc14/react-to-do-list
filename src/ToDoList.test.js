import { render, fireEvent, getByRole } from "@testing-library/react";
import ToDoList from "./ToDoList";

// smoke test
it("renders without crashing", function () {
  render(<ToDoList />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<ToDoList />);
  expect(asFragment()).toMatchSnapshot();
});

// functionality tests
it("can add a new box", function () {
  const { getByLabelText, queryByText } = render(<ToDoList />);

  // no boxes yet  (so no remove buttons)
  expect(queryByText("Test Item")).not.toBeInTheDocument();

  const nameInput = getByLabelText("Task:");
  const addBtn = queryByText("Add to List");

  // fill out the form:
  fireEvent.change(nameInput, { target: { value: "Test Item" } });
  fireEvent.click(addBtn);

  // item exists (remove button)
  expect(queryByText("Test Item")).toBeInTheDocument();
});
