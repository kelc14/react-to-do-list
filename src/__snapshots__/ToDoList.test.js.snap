// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoList"
  >
    <h2
      class="ToDoList-header"
    >
      My To Do List
    </h2>
    <div
      class="ToDoList-form"
    >
      <div
        class="ToDoForm"
      >
        <form
          class="ToDoForm-form"
        >
          <label
            for="name"
          >
            Task:
          </label>
          <input
            class="ToDoForm-input"
            id="name"
            name="name"
            type="text"
            value=""
          />
          <button
            class="ToDoForm-btn"
          >
            Add to List
          </button>
        </form>
      </div>
    </div>
    <div
      class="ToDoList-list-container"
    />
  </div>
</DocumentFragment>
`;
